### PR TITLE
Add filetype to python.yaml

### PR DESCRIPTION
**Pull request by bullerbeck**
I propose to add the "filetype" lib

Please add the following dependency to the rosdep database.

## Package name:

filetype

## Package Upstream Source:

https://pypi.org/project/filetype/

## Purpose of using this:

This dependency is being used to infer file type and MIME type checking the magic numbers signature of a file or buffer. This is why I think it's valuable to be added to the rosdep database. 

Distro packaging links:

filetype-1.0.7-py2.py3-none-any.whl

- Debian: https://packages.debian.org/
  - REQUIRED
- Ubuntu: https://packages.ubuntu.com/
   - REQUIRED
- Fedora: https://src.fedoraproject.org/browse/projects/
  - IF AVAILABLE
- Arch: https://www.archlinux.org/packages/
  - IF AVAILABLE
- Gentoo: https://packages.gentoo.org/
  - IF AVAILABLE
- macOS: https://formulae.brew.sh/
  - IF AVAILABLE
- Alpine: https://pkgs.alpinelinux.org/packages
  - IF AVAILABLE
- NixOS/nixpkgs: https://search.nixos.org/packages
  - OPTIONAL

# Checks
 - [x] All packages have a declared license in the package.xml
 - [x] This repository has a LICENSE file
 - [x] This package is expected to build on the submitted rosdistro

### CODE REVIEW

**Review by audrow — APPROVED** [review]

File: rosdep/python.yaml [hunk 1]
```diff
@@ -1648,6 +1648,13 @@ python-fcn-pip: &migrate_eol_2025_04_30_python3_fcn_pip
     pip:
       depends: [liblapack-dev]
       packages: [fcn]
+python-filetype-pip:
```
Comment (ivanpauno) [inline rosdep/python.yaml:1651]: there only seems to be a python3 version of the package (https://pypi.org/project/filetype/), but this is a python 2 entry.
If you wanted to add this for python3, please use `python3-filetype-pip`,

**Comment by hidmic** [comment]
@bullerbeck friendly ping

File: rosdep/python.yaml (hunk 1 above)
Comment (bullerbeck) [reply rosdep/python.yaml]: Correct. Resolved

Comment (hidmic) [reply rosdep/python.yaml]: @bullerbeck it should be `python3-filetype-pip`.

**Comment by hidmic** [comment]
There seems to be a formatting issue.

File: rosdep/python.yaml [hunk 2]
```diff
@@ -6304,6 +6304,13 @@ python3-fastnumbers-pip:
     pip:
       packages: [fastnumbers]
 python3-fcn-pip: *migrate_eol_2025_04_30_python3_fcn_pip
+python-filetype-pip:
```
Comment (ivanpauno) [inline rosdep/python.yaml:6307]:
```suggestion
python3-filetype-pip:
```

?

Comment (ivanpauno) [reply rosdep/python.yaml:6307]: sorry, already commented above https://github.com/ros/rosdistro/pull/30574#discussion_r702168494

**Review by hidmic — APPROVED** [review]

**Review by ivanpauno — APPROVED** [review]

**Pull request merged**